lab_3_generate_by_ngrams/main.py

```diff
@@ -236,6 +366,19 @@ def _extract_n_grams(
 
         In case of corrupt input arguments, None is returned
         """
+        if not isinstance(encoded_corpus, tuple) or len(encoded_corpus) == 0:
+            return None
+
+        n_grams_list = []
+        for index, number in enumerate(encoded_corpus):
+            if index == len(encoded_corpus) - (self._n_gram_size - 1):
+                break
+            n_gram = [number]
+            for i in range(1, self._n_gram_size):
```

Comment:
Consider directly slicing the needed subsequence

Reply:
done